hamilton/driver.py

```diff
@@ -153,10 +153,11 @@ def execute(
         """Basic executor for a function graph. Does no task-based execution, just does a DFS
         and executes the graph in order, in memory."""
         memoized_computation = dict()  # memoized storage
-        nodes = [fg.nodes[node_name] for node_name in final_vars]
+        nodes = [fg.nodes[node_name] for node_name in final_vars if node_name in fg.nodes]
         fg.execute(nodes, memoized_computation, overrides, inputs, run_id=run_id)
         outputs = {
-            final_var: memoized_computation[final_var] for final_var in final_vars
+            final_var: memoized_computation.get(final_var, inputs.get(final_var))
+            for final_var in final_vars
```

Comment:
You should be passing in all but the inputs to upstream nodes, rather than relying on the graph to know about it IMO -- see comment below

Reply:
oh sure I can just put the inputs into memoized dict in the first place..